python/ngen_cal/src/ngen/cal/calibration_set.py

```diff
@@ -53,41 +57,22 @@ def output(self) -> 'DataFrame':
             This re-reads the output file each call, as the output for given calibration catchment changes
             for each calibration iteration. If it doesn't exist, should return None
         """
-        try:
-            #in this case, look for routed data
-            #this is really model specific, so not as generalizable the way this
-            #is coded right now =(
-            #would be better to hook this from the model object???
-            #read the routed flow at the eval_nexus
-            df = pd.read_hdf(self._output_file)
-            df.index = df.index.map(lambda x: 'wb-'+str(x))
-            df.columns = pd.MultiIndex.from_tuples(df.columns)
-
-            # TODO should contributing_catchments be singular??? assuming it is for now...
-            df = df.loc[self._eval_nexus.contributing_catchments[0].replace('cat', 'wb')]
-            self._output = df.xs('q', level=1, drop_level=False)
-            #This is a hacky way to get the time index...pass the time around???
-            tnx_file = list(Path(self._output_file).parent.glob("nex*.csv"))[0]
-            tnx_df = pd.read_csv(tnx_file, index_col=0, parse_dates=[1], names=['ts', 'time', 'Q']).set_index('time')
-            dt_range = pd.date_range(tnx_df.index[0], tnx_df.index[-1], len(self._output.index)).round('min')
-            self._output.index = dt_range
-            #this may not be strictly nessicary...I think the _evalutate will align these...
-            self._output = self._output.resample('1H').first()
-            self._output.name="sim_flow"
-            # self._output = read_csv(self._output_file, usecols=["Time", self._output_var], parse_dates=['Time'], index_col='Time', dtype={self._output_var: 'float64'})
-            # self._output.rename(columns={self._output_var:'sim_flow'}, inplace=True)
-            hydrograph = self._output
-
-        except FileNotFoundError:
-            print("{} not found. Current working directory is {}".format(self._output_file, os.getcwd()))
-            print("Setting output to None")
-            hydrograph = None
-        except Exception as e:
-            raise(e)
-        #if hydrograph is None:
-        #    raise(RuntimeError("Error reading output: {}".format(self._output_file)))
-        return hydrograph
-
+        # TODO should contributing_catchments be singular??? assuming it is for now...
+        # Call output hooks, take first non-none result provided from hooks (called in LIFO order of registration)
+        df = self._output_hook(id=self._eval_nexus.contributing_catchments[0].replace('cat', 'wb'))
+        if not df:
```

Comment:
We should be able to get rid of this once `self` is added to the `hookspec`s above.

Reply:
I'll test this out and see what happens.

Reply:
Still get back a list object
```
[2018-01-01 00:00:00        0.538083
2018-01-01 01:00:00      789.833000
2018-01-01 02:00:00     8354.108000
2018-01-01 03:00:00    13602.951000
2018-01-01 04:00:00    13904.544000
2018-01-01 05:00:00    13648.108000
2018-01-01 06:00:00    13262.755000
2018-01-01 07:00:00    12760.119000
2018-01-01 08:00:00    12311.113000
2018-01-01 09:00:00    11847.882000
2018-01-01 10:00:00    11454.423000
2018-01-01 11:00:00    11056.809000
2018-01-01 12:00:00    10716.239000
2018-01-01 13:00:00    10372.783000
2018-01-01 14:00:00    10075.482000
2018-01-01 15:00:00     9775.703000
2018-01-01 16:00:00     9513.855000
2018-01-01 17:00:00     9249.671000
2018-01-01 18:00:00     9017.308000
2018-01-01 19:00:00     8797.409000
2018-01-01 20:00:00     8575.065000
2018-01-01 21:00:00     8377.901000
2018-01-01 22:00:00     8178.397000
2018-01-01 23:00:00     8000.547400
Freq: h, Name: sim_flow, dtype: float64]
<class 'list'>
```

Reply:
That's super weird. I really don't think that should be happening and is worth us figuring out why it is returning a list instead of a `pd.Series` as expected.

Reply:
Create an issue and leave the workaround for now?

Reply:
Sounds good and like something I can look into once we get this in. Opened https://github.com/NOAA-OWP/ngen-cal/issues/135